### DISCUSSION TITLE
CLI: Allow `router.base` configuration as command line argument

### DISCUSSION THREAD

**Discussion by JamesMGreene** [nuxt/nuxt#17175] [Ideas]
### Describe the feature you'd like to request

When I want to do a production deploy of a static HTML export of a NuxtJS site to GitHub Pages, or as a piece of a larger site, I need to:

1. Add a `router.base` configuration of something like `"/repoName/"` to the `nuxt.config.js` file
   - **_But_** I don't want that extra segment present for local development
2. Execute `nuxt generate`

I'm seeking a way to more easily automate NuxtJS projects across my organization, especially without need to worry about their particular application setup (CommonJS, ESM, TypeScript, etc.). As such, I would like to eliminate the need for the manual editing of the configuration file in Step 1 by combining it into Step 2.

### Describe the solution you'd like

I would like to be able to pass the desired `router.base` configuration value as a command line argument to the necessary `nuxt` CLI commands, e.g. `nuxt generate --router-base=/repoName/`.

The `generate` command is the main one that matters [to me], but it would probably make sense for the `build` command to also accept that parameter. 🤔 🤷🏻‍♂️ 

### Describe alternatives you've considered

The only good alternative I've found is requiring all of the projects to use a consistent snippet to support conditional values, e.g. using an environment variable to modify the `router.base` value:

```js
module.exports = {
  router: {
    base: process.env.BASE_PATH || process.env.NODE_ENV === 'production' ? '/repoName/' : '/'
  }
}
```

Alas, that requires changes to all of the repositories' codebase as well as participation/enforcement to ensure they remain in that kind of an ideal pattern. 😕 

### Prior Art

- 11ty's `--pathprefix`: https://www.11ty.dev/docs/config/#deploy-to-a-subdirectory-with-a-path-prefix
- Jekyll's `--baseurl`: https://jekyllrb.com/docs/configuration/options/#build-command-options
- Hugo's `--baseURL`: https://gohugo.io/commands/hugo/
- Gatsby is _basically_ there with its `pathPrefix` configuration that **must** be activated via a command line flag (`--prefix-paths`) or environment variable presence (`PREFIX_PATHS=true`): https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/path-prefix/
- Proposed feature request for Next.js: https://github.com/vercel/next.js/discussions/38776
- etc.

**Comment by justindasilva** [comment]
I would second this request.

In fact, at this current time, even when I set the following:
```
router: {
  base: '/docs/'
},
```

`nuxt generate` does not seem to be aware of the `base` that I have set.
